enclave/core/sgx/exception.c

```diff
@@ -528,10 +542,22 @@ void oe_virtual_exception_dispatcher(
     }
     else
     {
-        /* The unknown exception type indicates a host signal request. Validate
-         * the states on the td to ensure that the thread is handling the
-         * host signal */
-        if (!oe_sgx_td_host_signal_registered(td, (int)td->host_signal) ||
+        /* If the enclave is in debug mode and the oe_exception_record
+         * has been set, simulate the page fault based on host-passed
+         * information. This can either happen if the enclave runs with
+         * SGX1 CPUs or set CapturePFGPExceptions=0 on SGX2 CPUs. */
+        if (is_enclave_debug_allowed() && oe_exception_record)
+        {
+            td->exception_code = OE_EXCEPTION_PAGE_FAULT;
```

Comment:
For an SGX1 CPU, how would the user know that this code path is getting executed? If the user is unaware, then if the user switched from debug to production, there would be silent behavior difference.

Reply:
I added warning message. Do you think it's sufficient?

```
2022-03-23T22:11:09+0000.257161Z [(H)WARN] tid(0x7f33c4014b80) | The enclave is configured with CapturePFGPExceptions=1 but the current CPU does not support the feature. The #PF simulation will be enabled (debug-mode only). To disable the simulation, setting CapturePFGPExceptions=0.
 [/home/mingwei/workspace/openenclave/host/sgx/create.c:oe_sgx_build_enclave:1009]
```

Reply:
Yes. That is helpful for the user.